tarpc/tests/service_functional.rs

```diff
@@ -136,6 +136,45 @@ async fn serde() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[cfg(all(feature = "serde-transport", feature = "unix", unix))]
+#[tokio::test]
+async fn serde_uds() -> anyhow::Result<()> {
+    use std::fs;
+    use tarpc::serde_transport;
+    use tokio_serde::formats::Json;
+
+    const SOCK: &str = "tarpc_serde.sock";
```

Comment:
What about using a [temp directory](https://doc.rust-lang.org/stable/std/env/fn.temp_dir.html#) as well, to ensure cleanup in case the test crashes in the middle?

```suggestion
    let mut sock = std::env::temp_dir();
    sock.push("tarpc_serde.sock");
```

Perhaps also append some random number to prevent collisions on the off chance that tests are running concurrently?

Reply:
See [this comment in another thread](https://github.com/google/tarpc/pull/380#discussion_r984657583).